tools/functional-tester/etcd-runner/main.go

```diff
@@ -212,6 +219,179 @@ func runRacer(eps []string, round int) {
 	doRounds(rcs, round)
 }
 
+func runWatcher(eps []string) {
+	// watcher for setting watch on prefixes and also handling the response on watches
+	type watcher struct {
+		prefix string
+		c      *clientv3.Client
+		wc     clientv3.WatchChan
+	}
+
+	var runningTime time.Duration = 60 // time for which operation should be performed
+	noOfPrefixes := 36                 // total number of prefixes which will be watched upon
+	watchPerPrefix := 10               // number of watchers per prefix
+	reqRate := 20                      // put request per second
+	keyPrePrefix := 100000             // max number of keyPrePrefixs for put operation
+
+	ctx := context.Background()
+	prefixKeysMap := make(map[string][]string) // map for holding prefixes and values
+	ws := make([]watcher, noOfPrefixes*watchPerPrefix)
+	// pre generate prefixes equal to noOfPrefixes
+	prefixes := generateUniqueKeys(5, noOfPrefixes)
+	// pre generate keys
+	keys := generateRandomKeys(10, keyPrePrefix)
+
+	var (
+		revision int64
+		wg       sync.WaitGroup
+		pr       *clientv3.PutResponse
+		gr       *clientv3.GetResponse
+		err      error
+	)
+
+	// assign watchPerPrefix times watchers on each prefix
+	index := 0
+	wg.Add(len(ws))
+	for i := 0; i < len(ws); {
```

Comment:
@xiang90 after doing the changes i am seeting one error on running the test multiple times:
grpc: timed out when dialing
is it due to some leaks from this side. Since after re-starting the portal, it works and again on re-run it gives the same error